go-controller/hack/update-codegen.sh

```diff
@@ -72,6 +85,8 @@ for crd in ${crds}; do
 
 done
 
+rm -rf github.com/
```

Comment:
we should be generating in place => so we know where the go-controller is right in our system so rest should just be generated in place whether GOPATH is set or not set doesn't matter... I still don't understand why we are creating an entirely new nested folder which we are removing in L80 and then again copying that over and then again removing this here. what's with all the rm -rf's :)

Reply:
this part is a bit ugly, but it looks like before release 1.30 inputs and outputs of the scripts are tightly related to the generated code. So when we provide file paths here, they are propagated to the generated code imports, and it all falls apart. The best we can do here is let it generate fake api path, copy it to the desired destination and tmp folder.
btw it seems like inputs are properly found regardless of your env setup.
The extra removal of specific CRD folder before copying is needed to ensure stale files that may be stored in that folder are cleaned up.